Sources/GraphRepresentation.swift

```diff
@@ -0,0 +1,240 @@
+                    } else if let _ = dest as? MMScreenActionNode {
+                        renderer.renderEdgeToScreenAction(
+                            src: src,
+                            dest: dest.name,
+                            label: edge.predicate?.predicateFormat)
+                    } else {
+                        return
+                    }
+
+
+                }
+            } else if let node = node as? MMScreenActionNode {
+                if let destName = node.nextNodeName {
+                    renderer.renderEdgeToScreenState(src: src, dest: destName, label: nil, isBackable: true)
+                }
+            }
+        }
+
+        renderer.end()
+        return renderer.stringValue()
+    }
+}
+
+/// The default implementation
+class DotRepresentation: GraphRepresentation {
+    var lines: [String] = []
+
+    var namedIDs: [String: String] = [:]
+    var idGenerator = 0
+
+    let actionColor = "lightblue"
+    let fontActionColor = "white"
+
+    public var fileExtension: String { return "dot" }
+
+    init() {
+    }
+
+    func begin() {
+        lines = []
+        append("digraph G {",
+               "fontsize=15;",
+               "font=Helvetica;",
+               "labelloc=t;",
+               "label=\"\"", "splines=true", "overlap=false", "rankdir = LR;",
+               "ratio = auto;",
+               "node [ shape = box ]"
+        )
+
+        renderLegend()
+    }
+
+    func stringValue() -> String {
+        return lines.joined(separator: "\n")
+    }
+
+    private func append(_ additional: String...) {
+        lines = lines + additional
+    }
+
+    private func id(for name: String) -> String {
+        if let id = namedIDs[name] {
+            return id
+        }
+
+        let id = "_\(idGenerator)"
+        idGenerator += 1
+        namedIDs[name] = id
+        return id
+    }
+
+
+    func renderScreenStateNode(name: String, isDismissedOnUse dismissOnUse: Bool) {
+        let id = self.id(for: name)
+        var styleCode = "label=\"\(name)\"; "
+        if dismissOnUse {
+            styleCode += "fillcolor=lightgray; color=gray; style=filled"
+        } else {
+            styleCode += "color=black"
+        }
+        append("\(id) [ \(styleCode) ];")
+    }
+
+    func renderScreenActionNode(name: String) {
+        let id = self.id(for: name)
+        append("\(id) [ label = \"\(name)\"; shape=egg; style=filled; color=\(actionColor); fillcolor=\(actionColor); fontcolor=\(fontActionColor); fontsize=10];")
+    }
+
+    func renderEdgeToScreenState(src: String, dest: String, label: String?, isBackable: Bool) {
+        var styleCode: String = "[ "
+
+        if let label = label {
+            styleCode += "label=\"\(label)\"; style=dashed"
+        } else {
+            styleCode += "style=solid"
+        }
+
+        if isBackable {
+            styleCode += "; dir=both; arrowtail=obox; arrowhead=normal"
```

Comment:
nit: this is fine right now, but seems fragile... would be nice to have a utility function that lets you pass in a bunch of `key=value` strings that returns them concatenated together with proper `;` placements and whitespace

Reply:
Great idea!

Reply:
:+1: